RecoEgamma/EgammaTools/src/HGCalShowerShapeHelper.cc

```diff
@@ -0,0 +1,310 @@
+#include "RecoEgamma/EgammaTools/interface/HGCalShowerShapeHelper.h"
+
+void HGCalShowerShapeHelper::initPerEvent(const edm::EventSetup &iSetup, const std::vector<reco::PFRecHit> &pfRecHits) {
+  //recHitTools_.getEventSetup(iSetup);
+
+  edm::ESHandle<CaloGeometry> geom;
+  iSetup.get<CaloGeometryRecord>().get(geom);
+  recHitTools_.setGeometry(*(geom.product()));
+
+  setPFRecHitPtrMap(pfRecHits);
+}
+
+void HGCalShowerShapeHelper::initPerObject(const std::vector<std::pair<DetId, float> > &hitsAndFracs,
+                                           double minHitE,
+                                           double minHitET,
+                                           int minLayer,
+                                           int maxLayer,
+                                           DetId::Detector subDet) {
+  // Safety checks
+  nLayer_ = maxLayer - minLayer + 1;
+  assert(nLayer_ > 0);
```

Comment:
asserts are not the best way to achieve this.  If you really want to shut things down I would throw an exception. Still you could see what the CMSSW reviews say. They might have different opinions

Reply:
Okay, I'll wait for the review comments then.